libc/unistd_getopt.c

```diff
@@ -39,8 +39,7 @@ struct getopt_t {
 };
 
 
-int testargc = 0;
-struct getopt_t ret;
+static struct getopt_t ret;
```

Comment:
thanks for adding `static`, IMHO `ret` could be a local variable in tests (small enough to fit comfortably on stack on every platform), initialized to `0`, eg:
`struct getopt_t ret = { 0 };`

`testmain` could also be static, and `argv` `const` below